app/models/asset.rb

```diff
@@ -119,9 +119,9 @@ def accessible_by?(user)
 protected
 
   def store_metadata
-    self[:etag] ||= etag_from_file
-    self[:last_modified] ||= last_modified_from_file
-    self[:md5_hexdigest] ||= md5_hexdigest_from_file
+    self[:etag] = etag_from_file
+    self[:last_modified] = last_modified_from_file
+    self[:md5_hexdigest] = md5_hexdigest_from_file
```

Comment:
Is it important to use, for example, `self#[:md5_hexdigest]` instead of `self.md5_hexdigest=` in here?

Reply:
No, I don't think so. I've addressed this in a new commit, because this code is already in `master`.